velox/exec/tests/PrefixSortAlgorithmTest.cpp

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <folly/Random.h>
+#include <folly/String.h>
+#include <glog/logging.h>
+#include <gtest/gtest.h>
+#include <velox/exec/PrefixSortAlgorithm.h>
+#include <velox/vector/tests/VectorTestUtils.h>
+
+namespace facebook::velox::exec::test {
+
+namespace {
+void static printVector(std::vector<long>& vec) {
+  std::cout << folly::join(' ', vec) << std::endl;
+}
+
+int static compareStdVectorData(
+    const PrefixSortIterator& a,
+    const PrefixSortIterator& b) {
+  long v1 = *reinterpret_cast<long*>(*a);
+  long v2 = *reinterpret_cast<long*>(*b);
+  if (v1 == v2) {
+    return 0;
+  }
+  return v1 > v2 ? 1 : -1;
+}
+} // namespace
+
+class PrefixSortAlgorithmTest : public testing::Test,
+                                public velox::test::VectorTestBase {
+ public:
+  void testPrefixQuickSort(size_t size) {
+    // Data1 will be sorted by quickSort
+    std::vector<long> data1(size);
+    std::generate(
+        data1.begin(), data1.end(), [&]() { return folly::Random::rand64(); });
+    // Data2 will be sorted by std::sort
+    std::vector<long> data2 = data1;
+    char* start = (char*)data1.data();
+    char* end = start + sizeof(long) * data1.size();
+    uint32_t entrySize = sizeof(long);
+    auto context = PrefixSortContext(pool(), entrySize, end);
+    auto beginIter = PrefixSortIterator(start, entrySize);
+    auto endIter = beginIter + data1.size();
+    context.quickSort(beginIter, endIter, compareStdVectorData);
+    std::sort(data2.begin(), data2.end());
+    ASSERT_EQ(data1, data2);
+    printVector(data1);
```

Comment:
Test should not produce output. Please, remove.

For debugging, you can temporarily add LOG(ERROR << folly::join(", ", data1)